### PR TITLE
Add unit test for VSO bug 176584

### PR DESCRIPTION
**Pull request by sejongoh**

### CODE REVIEW

**Comment by sejongoh** [comment]
@RussKeldorph Could you please review this?

File: tests/src/JIT/Regression/JitBlue/vso_176584/vso_176584.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+static class MicrosftNETBug
```
Comment (RussKeldorph) [inline tests/src/JIT/Regression/JitBlue/vso_176584/vso_176584.cs]: Make sure you run this through the codeformatter, which I believe will also add the standard comment header.

File: tests/src/JIT/Regression/JitBlue/vso_176584/vso_176584.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    Console.WriteLine(".NET 4.6.01055 bug repro: second line incorrectly outputs False (only) when code is optimized");
+    
+    C c3 = new C(0);
+    C c1 = new C(256 * 128 - 1);
+    C c2 = new C(256 * 128);
+
+    Console.WriteLine(c1.D > c3.D);  
+    Console.WriteLine(c2.D > c3.D);            /* Gives false when optimization is enabled which is not correct */
+  }
```
Comment (RussKeldorph) [inline tests/src/JIT/Regression/JitBlue/vso_176584/vso_176584.cs]: I thought tests were supposed to return 100 to indicate success and others indicate failure.  Is there some other test harness magic making this pass that I'm not aware of?  (Wouldn't be surprised)  But if not, I think we may need an `if` to control the result.  Obviously may be easier or harder depending on the sensitivity of the bug to specific code patterns.

**Comment by RussKeldorph** [comment]
Other than the above, LGTM.

File: tests/src/JIT/Regression/JitBlue/vso_176584/vso_176584.cs (hunk 1 above)
Comment (sejongoh) [inline tests/src/JIT/Regression/JitBlue/vso_176584/vso_176584.cs]: I ran codeformatter and I added the header.

File: tests/src/JIT/Regression/JitBlue/vso_176584/vso_176584.cs (hunk 2 above)
Comment (sejongoh) [inline tests/src/JIT/Regression/JitBlue/vso_176584/vso_176584.cs]: The bug is sensitive to code pattern (no memory operand). I will try to add "if" and check if we can still see the expected code pattern.

**Comment by jasonwilliams200OK** [comment]
Please add some description about what was in VSO bug 1716584.

File: tests/src/JIT/Regression/JitBlue/vso_176584/vso_176584.cs (hunk 2 above)
Comment (sejongoh) [inline tests/src/JIT/Regression/JitBlue/vso_176584/vso_176584.cs]: I added "if" and can still see the code pattern.

**Comment by sejongoh** [comment]
@dotnet-bot test Windows_NT x86 Release Build

**Comment by sejongoh** [comment]
@dotnet-bot test Windows_NT x86 Debug Build

**Comment by sejongoh** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by sejongoh** [comment]
@dotnet-bot test Windows_NT x86 Release Build

**Comment by RussKeldorph** [comment]
LGTM

**Comment by sejongoh** [comment]
This testcase was renamed and committed by another PR.

**Pull request closed without merging**